Packages/com.vrchat.ClientSim/Runtime/CameraStacking/ClientSimStackedVRCameraSystem.cs

```diff
@@ -0,0 +1,121 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.SceneManagement;
+using UnityEngine.XR;
+using VRC.Core;
+using VRC.SDK3.ClientSim;
+
+namespace VRC.SDK3.ClientSim
+{
+    public class ClientSimStackedVRCameraSystem : MonoBehaviour
+    {
+        public ClientSimStackedCamera[] CameraStack;
+        [HideInInspector] public bool IsCameraStackingEnabled = false;
+        
+        private Camera MainSceneCamera;
+        private bool IsInitialized = false;
+        private bool IsReady = false;
+        private List<Camera> Cameras;
+        
+        private ClientSimMenu clientSimMenu;
+
+        public void Initialize(Camera playerCamera, ClientSimMenu menu)
+        {
+            MainSceneCamera = playerCamera;
+            clientSimMenu = menu;
+        }
+
+        public void Ready()
+        {
+            IsReady = true;
+        }
+        
+        void Update()
+        {
+            if(!IsReady) return;
+            if (!IsInitialized) { InitializeStackedSystem(); }
+            else
+            {
+                if (!Cameras[0].enabled)
+                {
+                    Debug.LogError("Stacked Cameras are not enabled.");
+                }
+            }
+        }
+     
+        void OnDisable()
+        {
+            if (MainSceneCamera != null)
+            {
+                if (IsCameraStackingEnabled)
+                    DestroyCameraStack();
+            }
+        }
+
+        public void InitializeStackedSystem()
+        {
+            Cameras = new List<Camera>();
+            if (MainSceneCamera != null)
+            {
+                CreateCameraStack();
+                IsInitialized = true;
+            }
+        }
+
+        private void CreateCameraStack()
+        {
+            for (int i = 0; i < CameraStack.Length; i++)
+            {
+                AddCamera(i);
+            }
+            IsCameraStackingEnabled = true;
+        }
+
+        private void DestroyCameraStack()
+        {
+            for (int i = 0; i < Cameras.Count; i++)
+            {
+                DestroyCamera(i);
+            }
+            IsCameraStackingEnabled = false;
+        }
+
+        private void AddCamera(int index)
+        {
+            GameObject cameraObj = Instantiate(new GameObject(), MainSceneCamera.transform);
+            Camera cam = cameraObj.AddComponent<Camera>();
+            XRDevice.DisableAutoXRCameraTracking(cam, true);
+
+            cam.CopyFrom(MainSceneCamera); // Start by copying all the settings from the main camera
+            #if VRC_VR_STEAM // We only want this on SteamVR.
+            cameraObj.AddComponent<SteamVRCantedProjectionCullingFix>();
+            #endif
```

Comment:
Do we usually indent preprocessor directives?
```suggestion
#if VRC_VR_STEAM // We only want this on SteamVR.
            cameraObj.AddComponent<SteamVRCantedProjectionCullingFix>();
#endif
```

Reply:
we are inconsistent about it, but rider prefers not indenting, so I'm happy to make this change

Reply:
Gotcha, up to you! Feel free to resolve this without committing.